### ISSUE TITLE
TensorFlow 2.0 plans

### ISSUE THREAD

**Issue by nredell** [vankesteren/tensorsem#4]
Hi. The tensorsem paper that you put out was an interesting read. Do you have any plans to port this package over to TensorFlow 2.0?

On a related note, do you have a roadmap or plan of what functionality you'd like to add next?

I'm curious because this seems like it'd be a worthwhile project to contribute to.

**Comment by vankesteren** [comment]
Hi Nick, 

thank you for the interest! Yes, I am currently looking at several features to move towards in the future. Tensorflow 2.0 is definitely high up on the to-do list, and more generally optimizing the tensorflow backend (e.g., looking at whether using GPU compute can speed up everything, seeing if any other optimizations can be made in [this file](https://github.com/vankesteren/tensorsem/blob/master/R/tf_pars_to_session.R)). Unfortunately right now I'm a bit busy with other projects and I'm not so very up-to-date on the whole tf1 - tf2 upgrade. I'm also not opposed to the idea of implementing the entire package in a different framework than tensorflow (torch, mxnet, or any other computation graph + autodiff + adam optimizer software) because I am not too sure about the whole `R -> python -> tensorflow` tower of cards -- but this requires quite some research.

If you know how to translate the current state of this package into tensorflow 2.0 I'd be very happy to receive a pull request! 

One thing on the feature roadmap is casewise / stochastic gradient descent optimization. There is a (kind of old) branch called `batch_processing` which implements this, but it's pretty slow and it uses some intransparent data wrangling for the `tfdatasets` package. 
Another thing I want to think about is how to let the user add their own tensors (penalties, alternative losses) to the computation graph without pre-programming them first as in [these lines](https://github.com/vankesteren/tensorsem/blob/3b82825ea3f528bcba7cc1d4d8790199837e042e/R/tf_pars_to_session.R#L98). Some kind of function pass-through in the `tf_sem()` function would be nice.

Ideally, there would be a CRAN release at some point too.

**Comment by nredell** [comment]
Bad follow-up on my part. Thanks for updating to version 2.0+. Looking forward to giving this a spin.
